### PR TITLE
Feature/orca 91 2

### PR DESCRIPTION
**Pull request by AndrewDorn**
## Summary of Changes

Please write a sentence or two summarizing the proposed change.

Addresses [request_files updated to provide/interact correctly with DB.](https://bugs.earthdata.nasa.gov/browse/ORCA-91)

## Changes

* request_files posts to new status queue.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the necessary documentation (remove those that do not apply)
    - [x] CHANGELOG.md
    - [ ] Testing documentation
    - [ ] Development documentation
    - [ ] User documentation
    - [ ] (Additional here)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
    - [x] Unit tests
    - [ ] Integration tests
    - [ ] Manual tests (outlined above)
- [x] New and existing unit tests pass locally with my changes
    - [x] Automated tests passing (if not fork)
    - [ ] Manual testing/integration testing passes (if forked)
- [x] Any dependent changes have been merged and published in downstream modules
- [x] My code has passed security scanning
    - [x] Snyk
    - [x] git-secrets

### CODE REVIEW

**Comment by AndrewDorn** [comment]
Remember to use updated queue code, and refresh documentation.

File: tasks/request_files/README.md [hunk 1]
```diff
@@ -23,7 +23,6 @@ Visit the [Developer Guide](https://nasa.github.io/cumulus-orca/docs/developer/d
     Edit the ['cumulus_meta']['execution_name'] to be something unique (like yyyymmdd_hhmm). Then
```
Comment (AndrewDorn) [inline tasks/request_files/README.md:23]: Forgot to update this.

**Review by bhazuka — CHANGES_REQUESTED** [review]
Minor change to CHANGELOG

File: tasks/request_files/schemas/input.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
+{
```
Comment (bhazuka) [inline tasks/request_files/schemas/input.json:1]: We may want to revisit this later after we do the archiving workflow functionality to see if we can get away with using one translator for both the archive and recovery path. I would maybe write a card on the backlog for investigation to do that analysis for future work.

File: tasks/request_files/schemas/output.json [hunk 3]
```diff
@@ -0,0 +1,51 @@
+{
```
Comment (bhazuka) [inline tasks/request_files/schemas/output.json:1]: We may want to revisit this later after we do the archiving workflow functionality to see if we can get away with a close to common output for both the archive and recovery path from the step function. I would maybe write a card on the backlog for investigation to do that analysis for future work.

File: tasks/request_files/schemas/input.json [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            "type": "string"
+          },
+          "keys": {
+            "description": "Keys related to where the files will be located.",
+            "type": "array",
+            "items": {
+              "type": "object",
+              "properties": {
+                "key": {
+                  "description": "Name of the file within the granule.  # TODO: This or example lies.",
```
Comment (bhazuka) [inline tasks/request_files/schemas/input.json:29]: I think the TODO is in the wrong spot. I feel it should be at the array of granules level. I believe only 1 granule is ever passed but a granule can have multiple files (keys). I could be wrong though. Maybe not possible with this first go around but we should maybe look at normalizing some of the key names in future work like keys -> files for example.

**Review by bhazuka — CHANGES_REQUESTED** [review]
A few minor items. Wondering if tasks/request_files/test/large_tests/test_request_files_postgres.py is still needed since no PostgreSQL work with this task anymore. Overall looking good. I may have missed it but, may want to create a bin directory and add our standard build/run_test.sh scripts to start to align with standards or at least create a card for future work to do that.

You will also need to update the Terraform code for this Lambda as you will need the environment variable changes.

File: CHANGELOG.md [hunk 5]
```diff
@@ -15,6 +15,7 @@ and includes an additional section for migration notes.
 
 
 ## [Unreleased]
+- *ORCA-91* copy_files_to_archive now uses SQS queue for recovery status updates. Will generate a job_id if none is given, and return it in the output.
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply CHANGELOG.md:18]: Fixed.

File: tasks/request_files/request_files.py [hunk 6]
```diff
@@ -2,21 +2,29 @@
...
-import requests_db
 # noinspection PyPackageRequirements
 from botocore.client import BaseClient
 # noinspection PyPackageRequirements
 from botocore.exceptions import ClientError
 from cumulus_logger import CumulusLogger
 from run_cumulus_task import run_cumulus_task
 
+
+class RequestMethod(Enum):
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply tasks/request_files/request_files.py:23]: Added

File: tasks/request_files/request_files.py [hunk 7]
```diff
@@ -54,6 +54,11 @@
 FILE_SUCCESS_KEY = 'success'
 FILE_ERROR_MESSAGE_KEY = 'err_msg'
 
+ORCA_STATUS_PENDING = 1
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply tasks/request_files/request_files.py:57]: Added

**Comment by AndrewDorn** [comment]
> build/run_test.sh



> A few minor items. Wondering if tasks/request_files/test/large_tests/test_request_files_postgres.py is still needed since no PostgreSQL work with this task anymore. Overall looking good. I may have missed it but, may want to create a bin directory and add our standard build/run_test.sh scripts to start to align with standards or at least create a card for future work to do that.
> 
> You will also need to update the Terraform code for this Lambda as you will need the environment variable changes.

Backlog item for switching how we run tests created as ORCA-199.

File: tasks/request_files/request_files.py [hunk 8]
```diff
@@ -137,15 +144,21 @@ def task(event: Dict, context: object) -> Dict[str, Any]:  # pylint: disable-msg
...
-    return inner_task(event, max_retries, retry_sleep_secs, retrieval_type, exp_days)
+    if not event[EVENT_INPUT_KEY].keys().__contains__(INPUT_JOB_ID_KEY):
+        event[EVENT_INPUT_KEY][INPUT_JOB_ID_KEY] = uuid.uuid4().__str__()
+
+    return inner_task(event, max_retries, retry_sleep_secs, retrieval_type, exp_days, db_queue_url)
 
 
-def inner_task(event: Dict, max_retries: int, retry_sleep_secs: float, retrieval_type: str, restore_expire_days: int):
+def inner_task(event: Dict, max_retries: int, retry_sleep_secs: float,
+               retrieval_type: str, restore_expire_days: int, db_queue_url: str):
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply tasks/request_files/request_files.py:163]: Moved stuff around.

File: tasks/request_files/request_files.py [hunk 9]
```diff
@@ -137,15 +144,21 @@ def task(event: Dict, context: object) -> Dict[str, Any]:  # pylint: disable-msg
...
     try:
         glacier_bucket = event[EVENT_CONFIG_KEY][CONFIG_GLACIER_BUCKET_KEY]
     except KeyError:
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply tasks/request_files/request_files.py:166]: I guess? Does this environment variable exist somewhere?

File: tasks/request_files/request_files.py [hunk 10]
```diff
@@ -198,36 +221,56 @@ def process_granule(s3: BaseClient, granule: Dict[str, Union[str, List[Dict]]],
...
-                    'days': restore_expire_days  # This property isn't from anything besides this code.
-                }
                 try:
-                    restore_object(s3, obj, attempt, max_retries, retrieval_type)
+                    restore_object(s3, a_file[FILE_KEY_KEY], restore_expire_days, glacier_bucket, attempt, job_id,
+                                   retrieval_type)
                     a_file[FILE_SUCCESS_KEY] = True
                     a_file[FILE_ERROR_MESSAGE_KEY] = ''
+
+                    post_status_for_file_to_queue(
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply tasks/request_files/request_files.py:260]: We could do something like that, but I'd prefer to save thinking about this until we have the new post code that posts for job+file simultaneously.

File: tasks/request_files/request_files.py [hunk 11]
```diff
@@ -237,11 +280,29 @@ def process_granule(s3: BaseClient, granule: Dict[str, Union[str, List[Dict]]],
...
+                db_queue_url,
+                max_retries,
+                retry_sleep_secs)
+
+    if any_error:
+        LOGGER.error(f"One or more files failed to be requested from {glacier_bucket}.{granule}")
+        raise RestoreRequestError(f'One or more files failed to be requested. {granule}')
 
 
 def object_exists(s3_cli: BaseClient, glacier_bucket: str, file_key: str) -> bool:
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply tasks/request_files/request_files.py:310]: I don't understand... Do you mean the function as a whole? That's called in inner_task. Do you mean the particular 'if' block you highlighted? Then it's called after the 'for' loop above it completes.

[the remark this answers is not in the document]
Comment (AndrewDorn) [reply tasks/request_files/request_files.py:310]: I don't know if this is the right place to enforce exclued_file_types... If we do that here, it'll still wind up in the status entries unless we do some extra checks/processing. I'd vote that the files in question should never be passed into process_granule in the first place, and should be checked/logged earlier.

File: tasks/request_files/requirements-dev.txt [hunk 12]
```diff
@@ -1,7 +1,7 @@
 cumulus-message-adapter-python>=1.1.1
-../dr_dbutils/dist/dr_dbutils-1.0.tar.gz
 ../pg_utils/dist/pg_utils-1.0.tar.gz
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply tasks/request_files/requirements-dev.txt]: Done

File: tasks/request_files/test/large_tests/test_request_files_postgres.py [hunk 13]
```diff
@@ -3,24 +3,21 @@
...
+
 import boto3
-from botocore.exceptions import ClientError
-from cumulus_logger import CumulusLogger
 import requests_db
-import db_config
+from cumulus_logger import CumulusLogger
 
-from test.request_helpers import LambdaContextMock, create_handler_event
+import db_config
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply tasks/request_files/test/large_tests/test_request_files_postgres.py:15]: File deleted.

File: tasks/request_files/test/large_tests/test_request_files_postgres.py [hunk 14]
```diff
@@ -3,24 +3,21 @@
...
 from test.request_helpers import (
     REQUEST_ID1, REQUEST_ID2, REQUEST_ID3, REQUEST_ID4,
-    REQUEST_GROUP_ID_EXP_1, REQUEST_GROUP_ID_EXP_2,
-    REQUEST_GROUP_ID_EXP_3, mock_secretsmanager_get_parameter)
-from test.request_helpers import print_rows
-
-import request_files
+    REQUEST_GROUP_ID_EXP_1, mock_secretsmanager_get_parameter)
 
 UTC_NOW_EXP_1 = requests_db.get_utc_now_iso()
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply tasks/request_files/test/large_tests/test_request_files_postgres.py:22]: File deleted.

File: tasks/request_files/schemas/input.json (hunk 2 above)
Comment (AndrewDorn) [reply tasks/request_files/schemas/input.json:1]: Do you mean translator like the lambdas that construct this input? Or CMA?

File: tasks/request_files/schemas/input.json (hunk 4 above)
Comment (AndrewDorn) [reply tasks/request_files/schemas/input.json:29]: Check out my todos in inner_task. Somebody wrote the code to loop over an array of granules, but then only ACTUALLY processes the last granule in the list. This particular todo is in regards to the example that uses a file path, despite this 'key' stating that it is a file name.

File: tasks/request_files/schemas/output.json (hunk 3 above)
Comment (AndrewDorn) [reply tasks/request_files/schemas/output.json:1]: Created, though I'm not entirely sure I'm parsing your request correctly. https://bugs.earthdata.nasa.gov/browse/ORCA-200

File: tasks/request_files/request_files.py (hunk 9 above)
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply tasks/request_files/request_files.py:195]: See copy_to_glacier

[the remark this answers is not in the document]
Comment (AndrewDorn) [reply tasks/request_files/request_files.py:195]: Adjust in code and Terraform

File: modules/orca/main.tf [hunk 15]
```diff
@@ -24,6 +24,32 @@ locals {
 
 ## Referenced Modules
 
+## orca_sqs - SQS module
```
Comment (AndrewDorn) [inline modules/orca/main.tf:27]: Undo

File: tasks/request_files/request_files.py (hunk 9 above)
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply tasks/request_files/request_files.py:203]: I think I got it.

File: modules/orca/main.tf [hunk 16]
```diff
@@ -22,8 +22,6 @@ locals {
 }
 
 
-## Referenced Modules
```
Comment (AndrewDorn) [inline modules/orca/main.tf:25]: Fix

File: modules/orca/main.tf [hunk 17]
```diff
@@ -95,6 +93,7 @@ module "orca_workflows" {
   orca_default_bucket                           = var.orca_default_bucket
   orca_lambda_extract_filepaths_for_granule_arn = module.orca_lambdas.extract_filepaths_for_granule_arn
   orca_lambda_request_files_arn                 = module.orca_lambdas.request_files_arn
+  orca_sqs_status_update_queue_id               = module.sqs.orca_sqs_status_update_queue_id
```
Comment (AndrewDorn) [inline modules/orca/main.tf:96]: Wrong module?

**Pull request merged**